opencog/query/Implicator.cc

```diff
@@ -49,8 +49,12 @@ bool Implicator::grounding(const HandleMap &var_soln,
 	if (_result_set.size() >= max_results)
 		return true;
 
-	Handle h = inst.instantiate(implicand, var_soln);
-	insert_result(h);
+	// By wrapping this in a try/catch we can ignore the cases when
+	// executions create ill-formed atoms.
+	try {
+		Handle h = inst.instantiate(implicand, var_soln);
+		insert_result(h);
+	} catch(const InvalidParamException& e) {}
```

Comment:
But if I change that to NotFound when the ImplicationLink fails line https://github.com/opencog/atomspace/blob/master/opencog/atoms/core/ImplicationLink.cc#L65 it will not catch it.